renku/api/storage.py

```diff
@@ -88,14 +88,18 @@ def track_paths_in_storage(self, *paths):
 
     def pull_path(self, path):
         """Pull a path from LFS."""
-        client, commit, path = self.resolve_in_submodules(
-            self.repo.commit(), path
-        )
-        run(
-            self._CMD_STORAGE_PULL + [str(path)],
-            cwd=str(client.path.absolute()),
-            check=True
-        )
+        if self.use_external_storage and self.external_storage_installed:
```

Comment:
Instead of branching we could just 

```
        if self.use_external_storage and self.external_storage_installed is False:
            raise errors.ExternalStorageNotInstalled(self.repo)
        ....

```

Easier to read for the future.

Reply:
we can refactor it to a decorator since the same pattern is in more methods.